### PR TITLE
WIP for new docs

### PR DESCRIPTION
**Pull request by victorpoluceno**
I've underestimated the amount of work to complete the docs. So, this is a WIP PR, as I will put this ticket back to the backlog and commit to finish this on next iteration.

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+With review environments, development teams can get quick feedback on their work by sharing a public url of the environment with it's peers.
+
+On top of that, environments can also be used for other purposes, like automated or manual testing.
+
+Environments can be deployed manually or through a CI/CD pipeline. Connecting the deployment to a CI/CD tool, makes possible to update the environment every time that a commit is pushed, which will speed up the feedback cycle.
+
+When deployed across an organization, review environment will be deployed in the same across all teams. This removes the confusion and the work of maintaining different review environments infrastructure, which is usually done in ad-hoc manner.
+
+### Infrascture
```
Comment (gtonello) [inline docs/introduction.md:17]: typo error

**Review by gtonello — APPROVED** [review]
Great work here. I agree with you that it's not "piece of cake", and could be split into some different tasks maybe. I left just one comment.

File: README.md [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
 
-1 - Choose the project development pipeline that you would like to test (e.g.: [geolocation-service-lab](https://g.codefresh.io/projects/geolocation-service-lab/edit/pipelines/?projectId=5fbf87e2b4b6c926b5fe6ebc))
+- Simple and universal deployment through any CI/CD tool;
+- Public accessible URL with HTTPS support;
+- Automatic removal of expired environments by time on when the branch is merged;
+- Complex environments with side car containers support;
+- Secrets and environment variables;
+- Ephemerals and Non-Ephemerals environments;
+- Custom environment names;
+- Kubernetes resource parametrization through Helm values file.
```
Comment (rlinstorres) [inline README.md:24]: You forget `;`

File: README.md [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
+- Customization through pre and post install hooks;
+- Scalable and lean infra using Nginx Ingress and Let's Encrypt WildCards domains;
+- Environment isolation through namespaces;
+- Connection test after deployment.
```
Comment (rlinstorres) [inline README.md:28]: You forget `;`

File: docs/developing.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+The docker image is hosted on [Docker Hub](https://hub.docker.com/r/findhotelamsterdam/kube-review).
+
+### Helm Chart
+
+The Helm Chart version uses `SemVer` for stable releases, but during development the version is always `0.0.1:{COMMIT_HASH}`.
+
+When the new chart version is created by the CI pipeline, it will push the new chart to the lab repo, using dev version schema, for example `0.0.1+87d6164`. To use this version you must pass it as the `KR_CHART_VERSION` variable.
+
+The Helm Chart is stored for stable releases is stored on the [main repo](cm://h.cfcr.io/findhotel/default/) and development releases at the [lab repo](cm://h.cfcr.io/findhotel/lab/).
```
Comment (rlinstorres) [inline docs/developing.md:23]: This is a public project, should we expose our helm repo on Codefresh or just use other information like an example? 🤔

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Getting Started
+
+As mentioned in the Introduction section, the first thing we need to deploy is a Kubernetes cluster with the necessary dependencies installed.
+
+## Kubernetes
+
+The first thing you need is a running Kubernetes cluster with a version >= 1.??. You can follow this either using your Kubernetes cluster provided by Docker or your own SaaS cluster like EKS.
```
Comment (rlinstorres) [inline docs/getting-started.md:7]: You can add `>= 1.16` it will be safer.

File: docs/introduction.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+Therefore, the Load Balance will send requests to the Nginx Ingress, which will route these requests to the correct container port based on the domain.
+
+In order to support HTTPS, Let's Encrypt is used as ACME to automatically handle certificate issuing and renewing.
+
+To use a well defined domain, users can choose any product for DNS resolution, we recommend AWS Route 53. Therefore, Route 53 will point the DNS entry to the AWS ELB.
+
+As we want to support a high number of review environments with HTTPS, we use WildCard domain certificates, in order to avoid hitting Let's Encrypt rate limiting.
+
+To manage the autoscaling and Nodes on your Kubernetes cluster in a cost effective way we recommend use of spot instances. You can use pure and simple AWS Spot Instances but we recommend the Spotinst service.
```
Comment (rlinstorres) [inline docs/introduction.md:31]: Advertising, `but we recommend the Spotinst service.`
Should we ask for a sponsorship? 😄

File: docs/reference.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+| KR_BRANCH_NAME | The branch that is getting deployed. This will be saved as annotation into the namespace so that the purge command can check the expiration of the branch/pr with the source code service. | - | false |
+| KR_REPO_NAME | The repo name source code in question. This will be saved as annotation into the namespace so that the purge command can check the expiration of the branch/pr with the source code service. | - | false |
+| KR_REPO_OWNER | The repository owner of the source code in question. This will be saved as annotation into the namespace so that the purge command can check the expiration of the branch/pr with the source code service.  | - | false |
+| KR_TEST_CONNECTION | Enable/disable testing the url of the environment once the deployment is done. If the connection fails the deployment will also fails. | true | false |
+| KR_PRE_HOOK | A shell command to be executed before the deployment starts. | - | false |
+| KR_POST_HOOK | A shell command to be executed after the deployment is finished. | - | false |
+| KR_HELM_REPO_URL | The helm repo url from which to get the app chart from. | cm://h.cfcr.io/findhotel/default/ | false |
```
Comment (rlinstorres) [inline docs/reference.md:31]: Just a note about our internal helm repository, if it is right or not.

**Review by rlinstorres — COMMENTED** [review]
I left small comments!

File: docs/introduction.md (hunk 6 above)
Comment (victorpoluceno) [reply docs/introduction.md:31]: Hahah, that would be awesome.

File: docs/getting-started.md (hunk 5 above)
Comment (victorpoluceno) [reply docs/getting-started.md:7]: Awesome, thanks.

File: docs/developing.md (hunk 4 above)
Comment (victorpoluceno) [reply docs/developing.md:23]: yes, we can. They are public repos, so no problem at all.

**Comment by victorpoluceno** [comment]
@rlinstorres @gtonello thanks, I will break this task into small stories and create an epic for it. I will use this PR as reference.

**Pull request closed without merging**